neuromation/api/config.py

```diff
@@ -365,6 +369,52 @@ def _check_sections(
             )
 
 
+def _check_item(
+    val: Any, validator: Any, full_name: str, filename: Union[str, "os.PathLike[str]"],
+) -> None:
+    if isinstance(validator, tuple):
+        # (container_type, item_type) pair
+        if not isinstance(val, validator[0]):
+            raise ConfigError(
+                f"{filename}: invalid type for {full_name}, "
+                f"{validator[0].__name__} is expected"
+            )
+        item_type = validator[1]
+        for num, i in enumerate(val):
+            _check_item(i, item_type, f"{full_name}[{num}]", filename)
+    else:
+        assert isinstance(validator, type) and issubclass(
+            validator, (bool, numbers.Real, numbers.Integral, str)
+        )
+        # validator for integer types should be numbers.Real or numbers.Integral,
+        # not int or float
+        if not isinstance(val, validator):
+            raise ConfigError(
+                f"{filename}: invalid type for {full_name}, "
+                f"{validator.__name__} is expected"
+            )
+
+
+def _check_section(
+    config: Mapping[str, Any],
+    section: str,
+    params: Dict[str, Any],
+    filename: Union[str, "os.PathLike[str]"],
+) -> None:
+    sec = config.get(section)
+    if sec is None:
+        return
+    diff = sec.keys() - params.keys()
+    if diff:
+        diff_str = ", ".join(f"{section}.{name}" for name in diff)
```

Comment:
```suggestion
        diff_str = ", ".join(f"{section}.{name}" for name in sorted(diff))
```
The order in a set is not defined.